scripts/appbuilder

```diff
@@ -158,27 +158,25 @@ get_image()
 # export a flattened copy of the container
 export_image()
 {
-    TEMP_FILE=$(mktemp)
     TEMP_DIR=$(mktemp -d)
     APPENV_FILENAME="appenv.json"
 
     # note that we have to start up the image to flatten it
     TEMP_INAME=$(docker run -d $IMAGE_NAME)
     TEMP_APPENV=$APPENV_FILENAME-$TEMP_INAME
-    
+
     docker image inspect --format='{{json .Config}}' $IMAGE_NAME > $TEMP_APPENV
 
     docker stop $TEMP_INAME >/dev/null
-    docker export $TEMP_INAME -o $TEMP_FILE
+    docker export $TEMP_INAME | tar xf - -C ${TEMP_DIR}
+
     docker rm $TEMP_INAME >/dev/null
     if [ -z $DELETE_IMAGE ]; then
 	docker rmi -f $IMAGE_NAME
     fi
 
-    # create the 'appdir' from the docker export
-    tar xf $TEMP_FILE -C $TEMP_DIR
     mv $TEMP_APPENV $TEMP_DIR/$APPENV_FILENAME
-    rm -rf $OUTPUT_NAME $TEMP_FILE
```

Comment:
It seems the change merged the 3 commands below using a pipe:
```
docker export -o tmp_file ... 
tar xf tmp_file
rm -rf tmp_file
``` 
The net effect is the same, the tmp_file is removed in both cases. Did I miss an optimization?

Reply:
That is correct but we are concerned about peek usage (when the tar file and the appdir exist at the same time). Peek usage is doubled without this.

A better optimization might be Anand's approach of mounting the docker image as a